Ical.Net.Tests/DeserializationTests.cs

```diff
@@ -612,4 +612,47 @@ public void CalendarWithMissingProdIdOrVersion_ShouldLeavePropertiesInvalid()
             Assert.That(deserialized, Does.Not.Contain("PRODID:").And.Not.Contains("VERSION:"));
         });
     }
+
+    [Test, Category("DurationSerializer")]
+    [TestCase("PT1H", 0, 0, 1, 0, 0)]
+    [TestCase("PT1H30M", 0, 0, 1, 30, 0)]
+    [TestCase("PT1H30M45S", 0, 0, 1, 30, 45)]
+    [TestCase("P1D", 0, 1, 0, 0, 0)]
+    [TestCase("-P1D", 0, -1, 0, 0, 0)]
+    [TestCase("P1W", 1, 0, 0, 0, 0)]
+    [TestCase("-P1W", -1, 0, 0, 0, 0)]
+    [TestCase("P0W", null, null, null, null, null)]
+    [TestCase("-P0W")]
+    [TestCase("-P1D", 0, -1, 0, 0, 0)]
+    [TestCase("-P1W", -1, 0, 0, 0, 0)]
+    [TestCase("PT0S", 0, 0, 0, 0, 0)]
+    [TestCase("-PT0S", 0, 0, 0, 0, 0)]
+    [TestCase("-PT1H", 0, 0, -1, 0, 0)]
+    [TestCase("-PT1H30M", 0, 0, -1, -30, 0)]
+    [TestCase("-P1D", 0, -1, 0, 0, 0)]
+    [TestCase("PT125H199M199S", 0, 0, 125, 199, 199)]
+    [TestCase("-PT125H199M199S", 0, 0, -125, -199, -199)]
+    [TestCase("-P0DT0H30M0S", 0, 0, 0, -30, 0)]
+    [TestCase("-P1DT1H", 0, -1, -1, 0, 0)]
+    [TestCase("P", null, null, null, null, null)] // Empty duration
+    [TestCase("PT", null, null, null, null, null)] // Empty time
```

Comment:
Is this legal?

Reply:
P and PT are not, I'll remove it for clarity